contrib/src/main/java/com/datatorrent/contrib/parser/CsvParser.java

```diff
@@ -62,247 +71,193 @@
-      } else if (type == FIELD_TYPE.BOOLEAN) {
-        processors[i] = new Optional(new ParseBool());
-      } else if (type == FIELD_TYPE.DATE) {
-        String dateFormat = getFields().get(i).format;
-        processors[i] = new Optional(new ParseDate(dateFormat == null ? "dd/MM/yyyy" : dateFormat));
-      }
-    }
-  }
   @Override
-  public Object convert(String tuple)
+  public Object convert(byte[] tuple)
   {
-    try {
-      csvStringReader.open(tuple);
-      return csvReader.read(clazz, nameMapping, processors);
-    } catch (IOException e) {
-      logger.debug("Error while converting tuple {} {}",tuple,e.getMessage());
-      return null;
-    }
+    //This method is not invoked for CSV parser
+    return null;
   }
 
   @Override
-  public void teardown()
+  public void processTuple(byte[] tuple)
   {
-    try {
-      if (csvReader != null) {
-        csvReader.close();
+    if (tuple == null) {
+      if (err.isConnected()) {
+        err.emit(new KeyValPair<String, String>(null, "Blank/null tuple"));
       }
-    } catch (IOException e) {
-      DTThrowable.rethrow(e);
-    }
-  }
-
-  @Override
-  public String processErorrTuple(String input)
-  {
-    return input;
-  }
-
-  public static class Field
-  {
-    String name;
-    String format;
-    FIELD_TYPE type;
-
-    public String getName()
-    {
-      return name;
+      errorTupleCount++;
+      return;
     }
-
-    public void setName(String name)
-    {
-      this.name = name;
-    }
-
-    public FIELD_TYPE getType()
-    {
-      return type;
-    }
-
-    public void setType(String type)
-    {
-      this.type = FIELD_TYPE.valueOf(type);
+    String incomingString = new String(tuple);
+    if (StringUtils.isBlank(incomingString) || StringUtils.equals(incomingString, header)) {
+      if (err.isConnected()) {
+        err.emit(new KeyValPair<String, String>(incomingString, "Blank/header tuple"));
+      }
+      errorTupleCount++;
+      return;
     }
+    try {
+      if (parsedOutput.isConnected()) {
+        csvStringReader.open(incomingString);
+        Map<String, Object> map = csvMapReader.read(nameMapping, processors);
+        parsedOutput.emit(map);
+        parsedOutputCount++;
+      }
 
-    public String getFormat()
-    {
-      return format;
-    }
+      if (out.isConnected() && clazz != null) {
+        csvStringReader.open(incomingString);
+        Object obj = csvBeanReader.read(clazz, nameMapping, processors);
+        out.emit(obj);
+        emittedObjectCount++;
+      }
 
-    public void setFormat(String format)
-    {
-      this.format = format;
+    } catch (SuperCsvException e) {
+      if (err.isConnected()) {
+        err.emit(new KeyValPair<String, String>(incomingString, e.getMessage()));
+      }
+      errorTupleCount++;
+      logger.error("Tuple could not be parsed. Reason {}", e.getMessage());
+    } catch (IOException e) {
+      logger.error("Exception in process method {}", e.getMessage());
```

Comment:
Why is IOException case distinguished from SuprtCsvException?

In case of SuperCsvException, we know for sure that it failed to parse, hence sent to err port.
In case of IOException do we know for sure that its not the case for sending to err port?